src/iexamine.cpp

```diff
@@ -4244,15 +4244,10 @@ static int findBestGasDiscount( player &p )
 {
     int discount = 0;
 
-    for( size_t i = 0; i < p.inv->size(); i++ ) {
-        item &it = p.inv->find_item( i );
-
-        if( it.has_flag( flag_GAS_DISCOUNT ) ) {
-
-            int q = getGasDiscountCardQuality( it );
-            if( q > discount ) {
-                discount = q;
-            }
+    for( const item *it : p.all_items_with_flag( flag_GAS_DISCOUNT ) ) {
+        int q = getGasDiscountCardQuality( *it );
+        if( q > discount ) {
+            discount = q;
         }
```

Comment:
Nitpick, but there's no need for introducing a new `q` variable which won't be used anywhere else. The whole snippet inside `for` cycle could be rewritten in one line: `std:max( discount, getGasDiscountCardQuality)`.

Reply:
Accepted.

Reply:
I will only note that this is not my code, I have not changed it.

Reply:
I understand, just a notice.